{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Modify Nodes in an NDEx Network Based on a TSV File"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this notebook, you will modify a network from NDEx that shows the interactions between the NCI ITCR projects.\n",
    "\n",
    "This version is missing some node attributes and needs corrections to the organization of their attributes.\n",
    "\n",
    "The network will be updated based on a TSV file where each row has information about one of the ITCR projects.\n",
    "\n",
    "The workflow:\n",
    " * Load the TSV file as a Pandas Dataframe\n",
    " * Create an NiceCX network from the network on NDEx\n",
    " * For each row in the Dataframe:\n",
    "  * Identify the corresponding node (skip if there is no such node)\n",
    "  * Add attributes to the node based on the following columns:\n",
    "   * “Category”\n",
    "   * “Tool URL”\n",
    "   * “Code Repo”\n",
    "   * “Project Info” \n",
    "  * Set the \"Tool Description\" attribute of the node to the value of the \"Body\" column\n",
    "  * Set the node name to be the value of the \"Title\" column\n",
    "  * Set the \"label\" attribute to the value of the \"Tool Acronym\" when available, otherwise use the value of the \"Title\" column.\n",
    " * Save the network to your account on NDEx\n",
    "\n",
    "The final network should still have 36 nodes/81 edges but provide extra info and link out to external resource"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, import the ndex2 Python client and Pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "import ndex2\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, set a variable to hold the UUID of the network we want to modify"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "my_network_uuid = '04c0a7e8-af92-11e7-94d3-0ac135e8bacf'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set variables to hold your account and password "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "my_account = \"drh\"\n",
    "my_password = \"drh\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create a NiceCX network from the ITCR network.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "downloading network and buiding NiceCX...\n",
      "done\n",
      "Name: ITCR Connectivity Map\n",
      "Nodes: 36\n",
      "Edges: 81\n",
      "Node Attributes: 0\n",
      "Edge Attributes: 241\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"downloading network and buiding NiceCX...\")\n",
    "itcr_network = ndex2.create_nice_cx_from_server(server='public.ndexbio.org', uuid=my_network_uuid)\n",
    "#my_network = ndex2.create_nice_cx_from_server(server='public.ndexbio.org', uuid=my_network_uuid, account=my_account, password = my_password)\n",
    "print(\"done\")\n",
    "print(itcr_network.get_summary())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the TSV to a Datframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Title</th>\n",
       "      <th>Tool Acronym</th>\n",
       "      <th>Body</th>\n",
       "      <th>Category</th>\n",
       "      <th>Tool URL</th>\n",
       "      <th>Code Repo</th>\n",
       "      <th>Project Info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3D Slicer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3D Slicer is the free open source software for...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>https://www.slicer.org</td>\n",
       "      <td>https://github.com/Slicer</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Allele-Specific Alternative mRNA processing (A...</td>\n",
       "      <td>ASARP</td>\n",
       "      <td>A software pipeline for prediction of allele-s...</td>\n",
       "      <td>omics</td>\n",
       "      <td>https://www.ibp.ucla.edu/research/xiao/Softwar...</td>\n",
       "      <td>https://github.com/cyruschan/ASARP</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Apache Clinical Text and Knowledge Extraction ...</td>\n",
       "      <td>cTAKES</td>\n",
       "      <td>The tool extracts deep phenotypic information ...</td>\n",
       "      <td>Clinical</td>\n",
       "      <td>http://ctakes.apache.org</td>\n",
       "      <td>http://ctakes.apache.org/downloads.cgi</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Bioconductor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bioconductor provides tools for the analysis a...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://www.bioconductor.org</td>\n",
       "      <td>http://bioconductor.org/developers/how-to/git/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Cancer and Phenomics Toolkit (CaPTK)</td>\n",
       "      <td>CaPTK</td>\n",
       "      <td>CaPTk is a tool that facilitates translation o...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://captk.projects.nitrc.org</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Cancer Slide Digital Archive (CDSA)</td>\n",
       "      <td>CDSA</td>\n",
       "      <td>The CDSA is a web-based platform to support th...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://cancer.digitalslidearchive.net</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Cancer-Related Analysis of Variants Toolkit (C...</td>\n",
       "      <td>CRAVAT</td>\n",
       "      <td>CRAVAT is an easy to use web-based tool for an...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://www.cravat.us/CRAVAT/</td>\n",
       "      <td>https://github.com/KarchinLab/CRAVAT</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Cistrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Curated and processed human/mouse ChIP/DNase-s...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://cistrome.org/Cistrome/Cistrome_Project....</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Clinical Interpretation of Variants in Cancer ...</td>\n",
       "      <td>CIViC</td>\n",
       "      <td>CIViC is an open access, open source, communit...</td>\n",
       "      <td>Omics, Clinical</td>\n",
       "      <td>https://civic.genome.wustl.edu/home</td>\n",
       "      <td>https://github.com/griffithlab</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Cores Of Recurrent Events (CORE)</td>\n",
       "      <td>CORE</td>\n",
       "      <td>CORE is a statistically supported computationa...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://cran.us.r-project.org/web/packages/CORE/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>D2Refine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D2Refine is an integrated platform that enable...</td>\n",
       "      <td>Data Standards</td>\n",
       "      <td>https://github.com/caCDE-QA/D2Refine/wiki</td>\n",
       "      <td>https://github.com/caCDE-QA/D2Refine</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DICOM ToolKit (DCMTK)</td>\n",
       "      <td>DCMTK</td>\n",
       "      <td>DCMTK is an open source DICOM toolkit.\\r\\n</td>\n",
       "      <td>Imaging, Data Standards</td>\n",
       "      <td>http://dicom.offis.de/dcmtk.php.en</td>\n",
       "      <td>http://git.dcmtk.org</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Enhancer Linking by Methylation/Expression Rel...</td>\n",
       "      <td>ELMER</td>\n",
       "      <td>R tool for analysis of DNA methylation and exp...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://bioconductor.org/packages/release/bioc/...</td>\n",
       "      <td>https://github.com/lijingya/ELMER</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Feature DB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The backend database to manage segmentation re...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/SBU-BMI/pathomics_featuredb</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Galaxy-P multiOmics</td>\n",
       "      <td>Galaxy-P</td>\n",
       "      <td>A unified platform for integrative genomic-pro...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://galaxyp.org</td>\n",
       "      <td>https://github.com/galaxyproteomics/tools-galaxyp</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GenePattern</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GenePattern is a genomic analysis platform tha...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://software.broadinstitute.org/cancer/soft...</td>\n",
       "      <td>https://github.com/genepattern/genepattern-server</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Genome-Independent Identification of RNA Editi...</td>\n",
       "      <td>GIREMI</td>\n",
       "      <td>Identification of single-nucleotide variants i...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/zhqingit/giremi</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Integrative Genomics Viewer (IGV)</td>\n",
       "      <td>IGV</td>\n",
       "      <td>The Integrative Genomics Viewer (IGV) is a hig...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://www.igv.org</td>\n",
       "      <td>https://github.com/igvteam</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>LSTM-CRF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LSTM-CRF uses Natural Language Processing meth...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/abhyudaynj/LSTM-CRF-models</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Methylation Integration (Mint)</td>\n",
       "      <td>Mint</td>\n",
       "      <td>A comprehensive software pipeline and set of G...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://github.com/sartorlab/mint</td>\n",
       "      <td>https://github.com/sartorlab/mint</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Modified UMS, Modified SemRep and SemMedDB-UTH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Modified UMLS, modified SemRep and SemMedDB-UT...</td>\n",
       "      <td>Clinical, Omics</td>\n",
       "      <td>https://skr3.nlm.nih.gov/SemMedDB/index_uth.html</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Mutated Positions in Annotated Structures (MuPIT)</td>\n",
       "      <td>MuPIT</td>\n",
       "      <td>MuPIT provides 3D visualization of mutation cl...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://mupit.icm.jhu.edu/MuPIT_Interactive/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>The Network Data Exchange (NDEx)</td>\n",
       "      <td>NDEx</td>\n",
       "      <td>NDEx is an online commons where scientists can...</td>\n",
       "      <td>Network Biology</td>\n",
       "      <td>http://www.ndexbio.org/</td>\n",
       "      <td>https://github.com/ndexbio</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Next Generation Clustered Heat Maps</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Next-Generation (Clustered) Heat Maps are inte...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://bioinformatics.mdanderson.org/TCGA/NGCH...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>OmniSearch</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OmniSearch is a semantic search software based...</td>\n",
       "      <td>Data Standards</td>\n",
       "      <td>http://omnisearch.soc.southalabama.edu/ui/</td>\n",
       "      <td>https://github.com/OmniSearch/ncro</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Ontology for MIcroRNA Target (OMIT)</td>\n",
       "      <td>OMIT</td>\n",
       "      <td>OMIT is a domain ontology specifically designe...</td>\n",
       "      <td>Data Standards</td>\n",
       "      <td>http://omnisearch.soc.southalabama.edu/w/index...</td>\n",
       "      <td>https://github.com/OmniSearch/ncro</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>LesionTracker Web Image Viewer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Extensible open-source zero-footprint web imag...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://lesiontracker.ohif.org</td>\n",
       "      <td>https://github.com/OHIF</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>owl-qa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>owl-qa is an OWL-based QA tool for cancer stud...</td>\n",
       "      <td>Data Standards</td>\n",
       "      <td>https://github.com/caCDE-QA/owl-qa</td>\n",
       "      <td>https://github.com/caCDE-QA/owl-qa</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>P-Mart</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Web-based software platform to enable analysis...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://pmart.labworks.org/index.html</td>\n",
       "      <td>https://hub.docker.com/r/pnnl/pmart-web/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Personalized Cancer Therapy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Curated website cataloguing clinically actiona...</td>\n",
       "      <td>Clinical, Omics</td>\n",
       "      <td>https://pct.mdanderson.org/#/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PhenStat</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PhenStat is a freely available R package that ...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://bioconductor.org/packages/release/bioc...</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Quantitative Imaging In Pathology (QUIP)</td>\n",
       "      <td>QUIP</td>\n",
       "      <td>This site hosts web accessible applications, t...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://quip.bmi.stonybrook.edu</td>\n",
       "      <td>https://github.com/SBU-BMI/nscale</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>RADIOMICS.io</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RADIOMICS.io is a open source platform for inf...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>https://www.radiomics.io</td>\n",
       "      <td>https://github.com/Radiomics</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Region Templates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A suite of libraries and runtime to support im...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/SBU-BMI/region-templates</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>SampleMap</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Projects samples into a GoogleMap explorer to ...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://tumormap.ucsc.edu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Pathology Imaging Informatics Platform (PIIP)</td>\n",
       "      <td>PIIP</td>\n",
       "      <td>Freely downloadable pathology viewer which sup...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://pathiip.org</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>FHIR ShEx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ShEx is a QA tool based on the Shape Expressio...</td>\n",
       "      <td>Data Standards</td>\n",
       "      <td>https://github.com/caCDE-QA/ShEX</td>\n",
       "      <td>https://github.com/caCDE-QA/ShEX</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>SlicerDMRI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diffusion magnetic resonance imaging for neuro...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://dmri.slicer.org</td>\n",
       "      <td>https://github.com/SlicerDMRI</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Text Information Extraction System (TIES)</td>\n",
       "      <td>TIES</td>\n",
       "      <td>TIES is a service based software system for ac...</td>\n",
       "      <td>Clinical</td>\n",
       "      <td>http://ties.dbmi.pitt.edu</td>\n",
       "      <td>https://sourceforge.net/projects/caties/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>The Laboratory for Individualized Breast Radio...</td>\n",
       "      <td>LIBRA</td>\n",
       "      <td>LIBRA is a fully-automatic breast density esti...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>https://www.med.upenn.edu/sbia/libra.html</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Trinity</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Trinity Cancer Transcriptome Analysis Toolkit ...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://github.com/trinityrnaseq/trinityrnaseq...</td>\n",
       "      <td>https://github.com/trinityrnaseq/trinityrnaseq</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>UCSC Xena</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UCSC Xena securely analyzes and visualizes you...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://xena.ucsc.edu</td>\n",
       "      <td>https://github.com/ucscXena/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>WebMeV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Web MeV (Multiple-experiment Viewer) is a web/...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://mev.tm4.org/#/welcome</td>\n",
       "      <td>https://github.com/dfci-cccb/mev</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>XNAT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XNAT is an open source imaging informatics pla...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://www.xnat.org</td>\n",
       "      <td>https://bitbucket.org/xnatdev/xnat-web</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Cloud-based Image Biomarker Platform  (C-BIBOP)</td>\n",
       "      <td>C-BIBOP</td>\n",
       "      <td>C-BIBOP is a cloud based platform for medical ...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://cbibop.github.io</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>ivyGlimpse</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bioconductor-based interface to Ivy-GAP (gliob...</td>\n",
       "      <td>Imaging, Omics</td>\n",
       "      <td>https://vjcitn.shinyapps.io/ivyglimpse/</td>\n",
       "      <td>https://github.com/vjcitn/ivygapSE</td>\n",
       "      <td>https://www.youtube.com/watch?v=4qJaodLEI5o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>The Cancer Proteome Atlas</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The Cancer Proteome Atals is a comprehensive b...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://tcpaportal.org</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>DINC 2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DINC 2.0 is a parallelized meta-docking method...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://dinc.kavrakilab.org</td>\n",
       "      <td>http://dinc.kavrakilab.org/help/</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>AMARETTO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AMARETTO for multiOmics data fusion and reform...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://personal.broadinstitute.org/npochet/re...</td>\n",
       "      <td>https://bitbucket.org/gevaertlab/pancanceramar...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>dcmqi: DICOM for Quantitative Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>dcmqi is a free, open source library that impl...</td>\n",
       "      <td>Data Standards, Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/qiicr/dcmqi</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Tools for Quantitative Analysis of PET Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A collection of extensions for 3D Slicer to en...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://qin.iibi.uiowa.edu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>CLAMP-Cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A tool to quickly build customized natural lan...</td>\n",
       "      <td>Clinical</td>\n",
       "      <td>http://clamp.uth.edu/cancer.php</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Tumor Heterogenity Research Interactive Visual...</td>\n",
       "      <td>THRIVE</td>\n",
       "      <td>A platform for quantitative evaluation of intr...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://www.csb.pitt.edu/ith/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>PETCT-CoSeg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A PET-CT co-segmentation tool for tumor deline...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/IOWA-PETCT-COSEG/PETCT_Slic...</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Single Cell Genome Viewer (SCGV)</td>\n",
       "      <td>SCGV</td>\n",
       "      <td>Graphics user interface for single-cell genomi...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/KrasnitzLab/SCGV</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>BioMuta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cancer associated gene mutation\\r\\n</td>\n",
       "      <td>Omics</td>\n",
       "      <td>https://hive.biochemistry.gwu.edu/biomuta</td>\n",
       "      <td>https://hive.biochemistry.gwu.edu/biomuta/readme</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>The Cancer Imaging Archive (TCIA)</td>\n",
       "      <td>TCIA</td>\n",
       "      <td>TCIA is NCI’s repository for publicly shared c...</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>http://www.cancerimagingarchive.net</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>cBioPortal for Cancer Genomics</td>\n",
       "      <td>cBioPortal</td>\n",
       "      <td>The cBioPortal for Cancer Genomics provides vi...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://www.cbioportal.org</td>\n",
       "      <td>https://github.com/cBioPortal/</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Patient-Derived Xenograft (PDX) Finder</td>\n",
       "      <td>PDX Finder</td>\n",
       "      <td>PDX Finder, currently in the prototyping phase...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://www.pdxfinder.org</td>\n",
       "      <td>https://github.com/PDXFinder/pdxfinder</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>The BioMedical Evidence Graph (BMEG)</td>\n",
       "      <td>BMEG</td>\n",
       "      <td>The BMEG is a Cancer Data integration Platform...</td>\n",
       "      <td>Omics</td>\n",
       "      <td>http://bmeg.io</td>\n",
       "      <td>https://github.com/bmeg</td>\n",
       "      <td>https://projectreporter.nih.gov/project_info_d...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Title Tool Acronym  \\\n",
       "0                                           3D Slicer          NaN   \n",
       "1   Allele-Specific Alternative mRNA processing (A...        ASARP   \n",
       "2   Apache Clinical Text and Knowledge Extraction ...       cTAKES   \n",
       "3                                        Bioconductor          NaN   \n",
       "4                Cancer and Phenomics Toolkit (CaPTK)        CaPTK   \n",
       "5                 Cancer Slide Digital Archive (CDSA)         CDSA   \n",
       "6   Cancer-Related Analysis of Variants Toolkit (C...       CRAVAT   \n",
       "7                                            Cistrome          NaN   \n",
       "8   Clinical Interpretation of Variants in Cancer ...        CIViC   \n",
       "9                    Cores Of Recurrent Events (CORE)         CORE   \n",
       "10                                           D2Refine          NaN   \n",
       "11                              DICOM ToolKit (DCMTK)        DCMTK   \n",
       "12  Enhancer Linking by Methylation/Expression Rel...        ELMER   \n",
       "13                                         Feature DB          NaN   \n",
       "14                                Galaxy-P multiOmics     Galaxy-P   \n",
       "15                                        GenePattern          NaN   \n",
       "16  Genome-Independent Identification of RNA Editi...       GIREMI   \n",
       "17                  Integrative Genomics Viewer (IGV)          IGV   \n",
       "18                                           LSTM-CRF          NaN   \n",
       "19                     Methylation Integration (Mint)         Mint   \n",
       "20     Modified UMS, Modified SemRep and SemMedDB-UTH          NaN   \n",
       "21  Mutated Positions in Annotated Structures (MuPIT)        MuPIT   \n",
       "22                   The Network Data Exchange (NDEx)         NDEx   \n",
       "23                Next Generation Clustered Heat Maps          NaN   \n",
       "24                                         OmniSearch          NaN   \n",
       "25                Ontology for MIcroRNA Target (OMIT)         OMIT   \n",
       "26                     LesionTracker Web Image Viewer          NaN   \n",
       "27                                             owl-qa          NaN   \n",
       "28                                             P-Mart          NaN   \n",
       "29                        Personalized Cancer Therapy          NaN   \n",
       "30                                           PhenStat          NaN   \n",
       "31           Quantitative Imaging In Pathology (QUIP)         QUIP   \n",
       "32                                       RADIOMICS.io          NaN   \n",
       "33                                   Region Templates          NaN   \n",
       "34                                          SampleMap          NaN   \n",
       "35      Pathology Imaging Informatics Platform (PIIP)         PIIP   \n",
       "36                                          FHIR ShEx          NaN   \n",
       "37                                         SlicerDMRI          NaN   \n",
       "38          Text Information Extraction System (TIES)         TIES   \n",
       "39  The Laboratory for Individualized Breast Radio...        LIBRA   \n",
       "40                                            Trinity          NaN   \n",
       "41                                          UCSC Xena          NaN   \n",
       "42                                             WebMeV          NaN   \n",
       "43                                               XNAT          NaN   \n",
       "44    Cloud-based Image Biomarker Platform  (C-BIBOP)      C-BIBOP   \n",
       "45                                         ivyGlimpse          NaN   \n",
       "46                          The Cancer Proteome Atlas          NaN   \n",
       "47                                           DINC 2.0          NaN   \n",
       "48                                           AMARETTO          NaN   \n",
       "49              dcmqi: DICOM for Quantitative Imaging          NaN   \n",
       "50     Tools for Quantitative Analysis of PET Imaging          NaN   \n",
       "51                                       CLAMP-Cancer          NaN   \n",
       "52  Tumor Heterogenity Research Interactive Visual...       THRIVE   \n",
       "53                                        PETCT-CoSeg          NaN   \n",
       "54                   Single Cell Genome Viewer (SCGV)         SCGV   \n",
       "55                                            BioMuta          NaN   \n",
       "56                 The Cancer Imaging Archive (TCIA)          TCIA   \n",
       "57                    cBioPortal for Cancer Genomics    cBioPortal   \n",
       "58            Patient-Derived Xenograft (PDX) Finder    PDX Finder   \n",
       "59              The BioMedical Evidence Graph (BMEG)          BMEG   \n",
       "\n",
       "                                                 Body  \\\n",
       "0   3D Slicer is the free open source software for...   \n",
       "1   A software pipeline for prediction of allele-s...   \n",
       "2   The tool extracts deep phenotypic information ...   \n",
       "3   Bioconductor provides tools for the analysis a...   \n",
       "4   CaPTk is a tool that facilitates translation o...   \n",
       "5   The CDSA is a web-based platform to support th...   \n",
       "6   CRAVAT is an easy to use web-based tool for an...   \n",
       "7   Curated and processed human/mouse ChIP/DNase-s...   \n",
       "8   CIViC is an open access, open source, communit...   \n",
       "9   CORE is a statistically supported computationa...   \n",
       "10  D2Refine is an integrated platform that enable...   \n",
       "11         DCMTK is an open source DICOM toolkit.\\r\\n   \n",
       "12  R tool for analysis of DNA methylation and exp...   \n",
       "13  The backend database to manage segmentation re...   \n",
       "14  A unified platform for integrative genomic-pro...   \n",
       "15  GenePattern is a genomic analysis platform tha...   \n",
       "16  Identification of single-nucleotide variants i...   \n",
       "17  The Integrative Genomics Viewer (IGV) is a hig...   \n",
       "18  LSTM-CRF uses Natural Language Processing meth...   \n",
       "19  A comprehensive software pipeline and set of G...   \n",
       "20  Modified UMLS, modified SemRep and SemMedDB-UT...   \n",
       "21  MuPIT provides 3D visualization of mutation cl...   \n",
       "22  NDEx is an online commons where scientists can...   \n",
       "23  Next-Generation (Clustered) Heat Maps are inte...   \n",
       "24  OmniSearch is a semantic search software based...   \n",
       "25  OMIT is a domain ontology specifically designe...   \n",
       "26  Extensible open-source zero-footprint web imag...   \n",
       "27  owl-qa is an OWL-based QA tool for cancer stud...   \n",
       "28  Web-based software platform to enable analysis...   \n",
       "29  Curated website cataloguing clinically actiona...   \n",
       "30  PhenStat is a freely available R package that ...   \n",
       "31  This site hosts web accessible applications, t...   \n",
       "32  RADIOMICS.io is a open source platform for inf...   \n",
       "33  A suite of libraries and runtime to support im...   \n",
       "34  Projects samples into a GoogleMap explorer to ...   \n",
       "35  Freely downloadable pathology viewer which sup...   \n",
       "36  ShEx is a QA tool based on the Shape Expressio...   \n",
       "37  Diffusion magnetic resonance imaging for neuro...   \n",
       "38  TIES is a service based software system for ac...   \n",
       "39  LIBRA is a fully-automatic breast density esti...   \n",
       "40  Trinity Cancer Transcriptome Analysis Toolkit ...   \n",
       "41  UCSC Xena securely analyzes and visualizes you...   \n",
       "42  Web MeV (Multiple-experiment Viewer) is a web/...   \n",
       "43  XNAT is an open source imaging informatics pla...   \n",
       "44  C-BIBOP is a cloud based platform for medical ...   \n",
       "45  Bioconductor-based interface to Ivy-GAP (gliob...   \n",
       "46  The Cancer Proteome Atals is a comprehensive b...   \n",
       "47  DINC 2.0 is a parallelized meta-docking method...   \n",
       "48  AMARETTO for multiOmics data fusion and reform...   \n",
       "49  dcmqi is a free, open source library that impl...   \n",
       "50  A collection of extensions for 3D Slicer to en...   \n",
       "51  A tool to quickly build customized natural lan...   \n",
       "52  A platform for quantitative evaluation of intr...   \n",
       "53  A PET-CT co-segmentation tool for tumor deline...   \n",
       "54  Graphics user interface for single-cell genomi...   \n",
       "55                Cancer associated gene mutation\\r\\n   \n",
       "56  TCIA is NCI’s repository for publicly shared c...   \n",
       "57  The cBioPortal for Cancer Genomics provides vi...   \n",
       "58  PDX Finder, currently in the prototyping phase...   \n",
       "59  The BMEG is a Cancer Data integration Platform...   \n",
       "\n",
       "                   Category  \\\n",
       "0                   Imaging   \n",
       "1                     omics   \n",
       "2                  Clinical   \n",
       "3                     Omics   \n",
       "4                   Imaging   \n",
       "5                   Imaging   \n",
       "6                     Omics   \n",
       "7                     Omics   \n",
       "8           Omics, Clinical   \n",
       "9                     Omics   \n",
       "10           Data Standards   \n",
       "11  Imaging, Data Standards   \n",
       "12                    Omics   \n",
       "13                  Imaging   \n",
       "14                    Omics   \n",
       "15                    Omics   \n",
       "16                    Omics   \n",
       "17                    Omics   \n",
       "18                    Omics   \n",
       "19                    Omics   \n",
       "20          Clinical, Omics   \n",
       "21                    Omics   \n",
       "22          Network Biology   \n",
       "23                    Omics   \n",
       "24           Data Standards   \n",
       "25           Data Standards   \n",
       "26                  Imaging   \n",
       "27           Data Standards   \n",
       "28                    Omics   \n",
       "29          Clinical, Omics   \n",
       "30                    Omics   \n",
       "31                  Imaging   \n",
       "32                  Imaging   \n",
       "33                  Imaging   \n",
       "34                    Omics   \n",
       "35                  Imaging   \n",
       "36           Data Standards   \n",
       "37                  Imaging   \n",
       "38                 Clinical   \n",
       "39                  Imaging   \n",
       "40                    Omics   \n",
       "41                    Omics   \n",
       "42                    Omics   \n",
       "43                  Imaging   \n",
       "44                  Imaging   \n",
       "45           Imaging, Omics   \n",
       "46                    Omics   \n",
       "47                    Omics   \n",
       "48                    Omics   \n",
       "49  Data Standards, Imaging   \n",
       "50                  Imaging   \n",
       "51                 Clinical   \n",
       "52                  Imaging   \n",
       "53                  Imaging   \n",
       "54                    Omics   \n",
       "55                    Omics   \n",
       "56                  Imaging   \n",
       "57                    Omics   \n",
       "58                    Omics   \n",
       "59                    Omics   \n",
       "\n",
       "                                             Tool URL  \\\n",
       "0                              https://www.slicer.org   \n",
       "1   https://www.ibp.ucla.edu/research/xiao/Softwar...   \n",
       "2                            http://ctakes.apache.org   \n",
       "3                        https://www.bioconductor.org   \n",
       "4                     http://captk.projects.nitrc.org   \n",
       "5               http://cancer.digitalslidearchive.net   \n",
       "6                        http://www.cravat.us/CRAVAT/   \n",
       "7   http://cistrome.org/Cistrome/Cistrome_Project....   \n",
       "8                 https://civic.genome.wustl.edu/home   \n",
       "9                                                 NaN   \n",
       "10          https://github.com/caCDE-QA/D2Refine/wiki   \n",
       "11                 http://dicom.offis.de/dcmtk.php.en   \n",
       "12  http://bioconductor.org/packages/release/bioc/...   \n",
       "13                                                NaN   \n",
       "14                                 http://galaxyp.org   \n",
       "15  http://software.broadinstitute.org/cancer/soft...   \n",
       "16                                                NaN   \n",
       "17                                 http://www.igv.org   \n",
       "18                                                NaN   \n",
       "19                  https://github.com/sartorlab/mint   \n",
       "20   https://skr3.nlm.nih.gov/SemMedDB/index_uth.html   \n",
       "21        http://mupit.icm.jhu.edu/MuPIT_Interactive/   \n",
       "22                            http://www.ndexbio.org/   \n",
       "23  http://bioinformatics.mdanderson.org/TCGA/NGCH...   \n",
       "24         http://omnisearch.soc.southalabama.edu/ui/   \n",
       "25  http://omnisearch.soc.southalabama.edu/w/index...   \n",
       "26                      http://lesiontracker.ohif.org   \n",
       "27                 https://github.com/caCDE-QA/owl-qa   \n",
       "28              https://pmart.labworks.org/index.html   \n",
       "29                      https://pct.mdanderson.org/#/   \n",
       "30                                                NaN   \n",
       "31                     http://quip.bmi.stonybrook.edu   \n",
       "32                           https://www.radiomics.io   \n",
       "33                                                NaN   \n",
       "34                          https://tumormap.ucsc.edu   \n",
       "35                                 http://pathiip.org   \n",
       "36                   https://github.com/caCDE-QA/ShEX   \n",
       "37                             http://dmri.slicer.org   \n",
       "38                          http://ties.dbmi.pitt.edu   \n",
       "39          https://www.med.upenn.edu/sbia/libra.html   \n",
       "40  https://github.com/trinityrnaseq/trinityrnaseq...   \n",
       "41                               http://xena.ucsc.edu   \n",
       "42                       http://mev.tm4.org/#/welcome   \n",
       "43                                http://www.xnat.org   \n",
       "44                            http://cbibop.github.io   \n",
       "45            https://vjcitn.shinyapps.io/ivyglimpse/   \n",
       "46                              http://tcpaportal.org   \n",
       "47                         http://dinc.kavrakilab.org   \n",
       "48  https://personal.broadinstitute.org/npochet/re...   \n",
       "49                                                NaN   \n",
       "50                          http://qin.iibi.uiowa.edu   \n",
       "51                    http://clamp.uth.edu/cancer.php   \n",
       "52                       http://www.csb.pitt.edu/ith/   \n",
       "53                                                NaN   \n",
       "54                                                NaN   \n",
       "55          https://hive.biochemistry.gwu.edu/biomuta   \n",
       "56                http://www.cancerimagingarchive.net   \n",
       "57                          http://www.cbioportal.org   \n",
       "58                           http://www.pdxfinder.org   \n",
       "59                                     http://bmeg.io   \n",
       "\n",
       "                                            Code Repo  \\\n",
       "0                           https://github.com/Slicer   \n",
       "1                  https://github.com/cyruschan/ASARP   \n",
       "2              http://ctakes.apache.org/downloads.cgi   \n",
       "3      http://bioconductor.org/developers/how-to/git/   \n",
       "4                                                 NaN   \n",
       "5                                                 NaN   \n",
       "6                https://github.com/KarchinLab/CRAVAT   \n",
       "7                                                 NaN   \n",
       "8                      https://github.com/griffithlab   \n",
       "9     http://cran.us.r-project.org/web/packages/CORE/   \n",
       "10               https://github.com/caCDE-QA/D2Refine   \n",
       "11                               http://git.dcmtk.org   \n",
       "12                  https://github.com/lijingya/ELMER   \n",
       "13     https://github.com/SBU-BMI/pathomics_featuredb   \n",
       "14  https://github.com/galaxyproteomics/tools-galaxyp   \n",
       "15  https://github.com/genepattern/genepattern-server   \n",
       "16                 https://github.com/zhqingit/giremi   \n",
       "17                         https://github.com/igvteam   \n",
       "18      https://github.com/abhyudaynj/LSTM-CRF-models   \n",
       "19                  https://github.com/sartorlab/mint   \n",
       "20                                                NaN   \n",
       "21                                                NaN   \n",
       "22                         https://github.com/ndexbio   \n",
       "23                                                NaN   \n",
       "24                 https://github.com/OmniSearch/ncro   \n",
       "25                 https://github.com/OmniSearch/ncro   \n",
       "26                            https://github.com/OHIF   \n",
       "27                 https://github.com/caCDE-QA/owl-qa   \n",
       "28           https://hub.docker.com/r/pnnl/pmart-web/   \n",
       "29                                                NaN   \n",
       "30  https://bioconductor.org/packages/release/bioc...   \n",
       "31                  https://github.com/SBU-BMI/nscale   \n",
       "32                       https://github.com/Radiomics   \n",
       "33        https://github.com/SBU-BMI/region-templates   \n",
       "34                                                NaN   \n",
       "35                                                NaN   \n",
       "36                   https://github.com/caCDE-QA/ShEX   \n",
       "37                      https://github.com/SlicerDMRI   \n",
       "38           https://sourceforge.net/projects/caties/   \n",
       "39                                                NaN   \n",
       "40     https://github.com/trinityrnaseq/trinityrnaseq   \n",
       "41                       https://github.com/ucscXena/   \n",
       "42                   https://github.com/dfci-cccb/mev   \n",
       "43             https://bitbucket.org/xnatdev/xnat-web   \n",
       "44                                                NaN   \n",
       "45                 https://github.com/vjcitn/ivygapSE   \n",
       "46                                                NaN   \n",
       "47                   http://dinc.kavrakilab.org/help/   \n",
       "48  https://bitbucket.org/gevaertlab/pancanceramar...   \n",
       "49                     https://github.com/qiicr/dcmqi   \n",
       "50                                                NaN   \n",
       "51                                                NaN   \n",
       "52                                                NaN   \n",
       "53  https://github.com/IOWA-PETCT-COSEG/PETCT_Slic...   \n",
       "54                https://github.com/KrasnitzLab/SCGV   \n",
       "55   https://hive.biochemistry.gwu.edu/biomuta/readme   \n",
       "56                                                NaN   \n",
       "57                     https://github.com/cBioPortal/   \n",
       "58             https://github.com/PDXFinder/pdxfinder   \n",
       "59                            https://github.com/bmeg   \n",
       "\n",
       "                                         Project Info  \n",
       "0   https://projectreporter.nih.gov/project_info_d...  \n",
       "1   https://projectreporter.nih.gov/project_info_d...  \n",
       "2   https://projectreporter.nih.gov/project_info_d...  \n",
       "3   https://projectreporter.nih.gov/project_info_d...  \n",
       "4   https://projectreporter.nih.gov/project_info_d...  \n",
       "5   https://projectreporter.nih.gov/project_info_d...  \n",
       "6   https://projectreporter.nih.gov/project_info_d...  \n",
       "7   https://projectreporter.nih.gov/project_info_d...  \n",
       "8   https://projectreporter.nih.gov/project_info_d...  \n",
       "9   https://projectreporter.nih.gov/project_info_d...  \n",
       "10  https://projectreporter.nih.gov/project_info_d...  \n",
       "11  https://projectreporter.nih.gov/project_info_d...  \n",
       "12  https://projectreporter.nih.gov/project_info_d...  \n",
       "13  https://projectreporter.nih.gov/project_info_d...  \n",
       "14  https://projectreporter.nih.gov/project_info_d...  \n",
       "15  https://projectreporter.nih.gov/project_info_d...  \n",
       "16  https://projectreporter.nih.gov/project_info_d...  \n",
       "17  https://projectreporter.nih.gov/project_info_d...  \n",
       "18  https://projectreporter.nih.gov/project_info_d...  \n",
       "19  https://projectreporter.nih.gov/project_info_d...  \n",
       "20  https://projectreporter.nih.gov/project_info_d...  \n",
       "21  https://projectreporter.nih.gov/project_info_d...  \n",
       "22  https://projectreporter.nih.gov/project_info_d...  \n",
       "23  https://projectreporter.nih.gov/project_info_d...  \n",
       "24  https://projectreporter.nih.gov/project_info_d...  \n",
       "25  https://projectreporter.nih.gov/project_info_d...  \n",
       "26  https://projectreporter.nih.gov/project_info_d...  \n",
       "27  https://projectreporter.nih.gov/project_info_d...  \n",
       "28  https://projectreporter.nih.gov/project_info_d...  \n",
       "29  https://projectreporter.nih.gov/project_info_d...  \n",
       "30  https://projectreporter.nih.gov/project_info_d...  \n",
       "31  https://projectreporter.nih.gov/project_info_d...  \n",
       "32  https://projectreporter.nih.gov/project_info_d...  \n",
       "33  https://projectreporter.nih.gov/project_info_d...  \n",
       "34  https://projectreporter.nih.gov/project_info_d...  \n",
       "35  https://projectreporter.nih.gov/project_info_d...  \n",
       "36  https://projectreporter.nih.gov/project_info_d...  \n",
       "37  https://projectreporter.nih.gov/project_info_d...  \n",
       "38  https://projectreporter.nih.gov/project_info_d...  \n",
       "39  https://projectreporter.nih.gov/project_info_d...  \n",
       "40  https://projectreporter.nih.gov/project_info_d...  \n",
       "41  https://projectreporter.nih.gov/project_info_d...  \n",
       "42  https://projectreporter.nih.gov/project_info_d...  \n",
       "43  https://projectreporter.nih.gov/project_info_d...  \n",
       "44  https://projectreporter.nih.gov/project_info_d...  \n",
       "45        https://www.youtube.com/watch?v=4qJaodLEI5o  \n",
       "46  https://projectreporter.nih.gov/project_info_d...  \n",
       "47                                                NaN  \n",
       "48                                                NaN  \n",
       "49  https://projectreporter.nih.gov/project_info_d...  \n",
       "50  https://projectreporter.nih.gov/project_info_d...  \n",
       "51                                                NaN  \n",
       "52  https://projectreporter.nih.gov/project_info_d...  \n",
       "53  https://projectreporter.nih.gov/project_info_d...  \n",
       "54  https://projectreporter.nih.gov/project_info_d...  \n",
       "55                                                NaN  \n",
       "56  https://projectreporter.nih.gov/project_info_d...  \n",
       "57  https://projectreporter.nih.gov/project_info_d...  \n",
       "58                                                NaN  \n",
       "59  https://projectreporter.nih.gov/project_info_d...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tsv_path = '../resources/itcr-conn-map-v2.0_nodes.txt'\n",
    "node_data = pd.read_csv(tsv_path, sep=\"\\t\")\n",
    "node_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Annotate the nodes in the network according to the plan.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\"r\": \"C-BIBOP\", \"@id\": 1, \"n\": \"C-BIBOP\"}\n",
      "{\"r\": \"3D Slicer\", \"@id\": 2, \"n\": \"3D Slicer\"}\n",
      "{\"r\": \"CaPTK\", \"@id\": 3, \"n\": \"CaPTK\"}\n",
      "{\"r\": \"DICOM ToolKit (DCMTK)\", \"@id\": 4, \"n\": \"DICOM ToolKit (DCMTK)\"}\n",
      "{\"r\": \"LesionTracker\", \"@id\": 5, \"n\": \"LesionTracker\"}\n",
      "{\"r\": \"PETCT-CoSeg\", \"@id\": 6, \"n\": \"PETCT-CoSeg\"}\n",
      "{\"r\": \"QUIP\", \"@id\": 7, \"n\": \"QUIP\"}\n",
      "{\"r\": \"SlicerDMRI\", \"@id\": 8, \"n\": \"SlicerDMRI\"}\n",
      "{\"r\": \"XNAT\", \"@id\": 9, \"n\": \"XNAT\"}\n",
      "{\"r\": \"GenePattern\", \"@id\": 10, \"n\": \"GenePattern\"}\n",
      "{\"r\": \"AMARETTO\", \"@id\": 11, \"n\": \"AMARETTO\"}\n",
      "{\"r\": \"MuPIT\", \"@id\": 12, \"n\": \"MuPIT\"}\n",
      "{\"r\": \"BMEG\", \"@id\": 13, \"n\": \"BMEG\"}\n",
      "{\"r\": \"NG-CHM\", \"@id\": 14, \"n\": \"NG-CHM\"}\n",
      "{\"r\": \"CDSA\", \"@id\": 15, \"n\": \"CDSA\"}\n",
      "{\"r\": \"dcmqi\", \"@id\": 16, \"n\": \"dcmqi\"}\n",
      "{\"r\": \"RADIOMICS.io\", \"@id\": 17, \"n\": \"RADIOMICS.io\"}\n",
      "{\"r\": \"Bioconductor\", \"@id\": 18, \"n\": \"Bioconductor\"}\n",
      "{\"r\": \"cBioPortal\", \"@id\": 19, \"n\": \"cBioPortal\"}\n",
      "{\"r\": \"NDEx\", \"@id\": 20, \"n\": \"NDEx\"}\n",
      "{\"r\": \"CIViC\", \"@id\": 21, \"n\": \"CIViC\"}\n",
      "{\"r\": \"HIVE (BioMuta)\", \"@id\": 22, \"n\": \"HIVE (BioMuta)\"}\n",
      "{\"r\": \"PDX Finder\", \"@id\": 23, \"n\": \"PDX Finder\"}\n",
      "{\"r\": \"Trinity\", \"@id\": 24, \"n\": \"Trinity\"}\n",
      "{\"r\": \"FHIR ShEx\", \"@id\": 25, \"n\": \"FHIR ShEx\"}\n",
      "{\"r\": \"CLAMP-Cancer\", \"@id\": 26, \"n\": \"CLAMP-Cancer\"}\n",
      "{\"r\": \"Galaxy-P multi-omics\", \"@id\": 27, \"n\": \"Galaxy-P multi-omics\"}\n",
      "{\"r\": \"CRAVAT\", \"@id\": 28, \"n\": \"CRAVAT\"}\n",
      "{\"r\": \"UCSC Xena\", \"@id\": 29, \"n\": \"UCSC Xena\"}\n",
      "{\"r\": \"Integrative Genomics Viewer (IGV)\", \"@id\": 30, \"n\": \"Integrative Genomics Viewer (IGV)\"}\n",
      "{\"r\": \"THRIVE\", \"@id\": 31, \"n\": \"THRIVE\"}\n",
      "{\"r\": \"OmniSearch\", \"@id\": 32, \"n\": \"OmniSearch\"}\n",
      "{\"r\": \"TCIA\", \"@id\": 33, \"n\": \"TCIA\"}\n",
      "{\"r\": \"HIVE (BioExpress)\", \"@id\": 34, \"n\": \"HIVE (BioExpress)\"}\n",
      "{\"r\": \"TCPA\", \"@id\": 35, \"n\": \"TCPA\"}\n",
      "{\"r\": \"TIES\", \"@id\": 36, \"n\": \"TIES\"}\n"
     ]
    }
   ],
   "source": [
    "label_to_node_map = {}\n",
    "for index, node in itcr_network.get_nodes():\n",
    "    label = itcr_network.get_node_attribute(node, \"label\")\n",
    "    if label:\n",
    "        label_to_node_map[label] = node\n",
    "        \n",
    "for index, row in node_data.iterrows():\n",
    "    # Identify the corresponding node (skip if there is no such node)\n",
    "    node = label_to_node_map.get(row['Title'])\n",
    "    if node:\n",
    "        \n",
    "        # Add attributes to the node\n",
    "        attributes = [\"Category\", \"Tool URL\", \"Code Repo\", \"Project Info\"]\n",
    "        for att in attributes:\n",
    "            itcr_network.set_node_attribute(node, att, row[att])\n",
    "            \n",
    "        # Set the \"Tool Description\" attribute of the node to the value of the \"Body\" column\n",
    "        itcr_network.set_node_attribute(node, \"Tool Description\", row['Body'])\n",
    "        \n",
    "        # Set the node name to be the value of the \"Title\" column\n",
    "        node.set_name(row[\"Title\"])\n",
    "        \n",
    "        # Set the \"label\" attribute to the value of the \"Tool Acronym\" when available, \n",
    "        # otherwise use the value of the \"Title\" column.\n",
    "        acronym = row[\"Tool Acronym\"]\n",
    "        if acronym is \"NaN\":\n",
    "            acronym = None\n",
    "        itcr_network.set_node_attribute(node, \"label\", acronym or row[\"Title\"])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Spot-check the network by printing out a sample of the nodes in the modified network"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[(1, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710310>), (2, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710050>), (3, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710290>), (4, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710250>), (5, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710150>), (6, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f7102d0>), (7, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710190>), (8, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f7103d0>), (9, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f710090>), (10, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb850>), (11, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebf90>), (12, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebbd0>), (13, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb550>), (14, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eba90>), (15, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb610>), (16, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb8d0>), (17, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb7d0>), (18, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb910>), (19, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb890>), (20, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebd90>), (21, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb650>), (22, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb810>), (23, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebe10>), (24, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebe90>), (25, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebf10>), (26, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb950>), (27, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eba50>), (28, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebf50>), (29, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb750>), (30, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebad0>), (31, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb690>), (32, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebed0>), (33, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6eb790>), (34, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebb10>), (35, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebb90>), (36, <nicecxModel.cx.aspects.NodeElement.NodeElement object at 0x11f6ebc10>)]\n"
     ]
    }
   ],
   "source": [
    "print(itcr_network.get_nodes())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C-BIBOP\n",
      "None\n",
      "None\n",
      "----\n",
      "3D Slicer\n",
      "None\n",
      "None\n",
      "----\n",
      "CaPTK\n",
      "None\n",
      "None\n",
      "----\n",
      "DICOM ToolKit (DCMTK)\n",
      "None\n",
      "None\n",
      "----\n",
      "LesionTracker\n",
      "None\n",
      "None\n",
      "----\n",
      "PETCT-CoSeg\n",
      "None\n",
      "None\n",
      "----\n",
      "QUIP\n",
      "None\n",
      "None\n",
      "----\n",
      "SlicerDMRI\n",
      "None\n",
      "None\n",
      "----\n",
      "XNAT\n",
      "None\n",
      "None\n",
      "----\n",
      "GenePattern\n",
      "None\n",
      "None\n",
      "----\n",
      "AMARETTO\n",
      "None\n",
      "None\n",
      "----\n",
      "MuPIT\n",
      "None\n",
      "None\n",
      "----\n",
      "BMEG\n",
      "None\n",
      "None\n",
      "----\n",
      "NG-CHM\n",
      "None\n",
      "None\n",
      "----\n",
      "CDSA\n",
      "None\n",
      "None\n",
      "----\n",
      "dcmqi\n",
      "None\n",
      "None\n",
      "----\n",
      "RADIOMICS.io\n",
      "None\n",
      "None\n",
      "----\n",
      "Bioconductor\n",
      "None\n",
      "None\n",
      "----\n",
      "cBioPortal\n",
      "None\n",
      "None\n",
      "----\n",
      "NDEx\n",
      "None\n",
      "None\n",
      "----\n",
      "CIViC\n",
      "None\n",
      "None\n",
      "----\n",
      "HIVE (BioMuta)\n",
      "None\n",
      "None\n",
      "----\n",
      "PDX Finder\n",
      "None\n",
      "None\n",
      "----\n",
      "Trinity\n",
      "None\n",
      "None\n",
      "----\n",
      "FHIR ShEx\n",
      "None\n",
      "None\n",
      "----\n",
      "CLAMP-Cancer\n",
      "None\n",
      "None\n",
      "----\n",
      "Galaxy-P multi-omics\n",
      "None\n",
      "None\n",
      "----\n",
      "CRAVAT\n",
      "None\n",
      "None\n",
      "----\n",
      "UCSC Xena\n",
      "None\n",
      "None\n",
      "----\n",
      "Integrative Genomics Viewer (IGV)\n",
      "None\n",
      "None\n",
      "----\n",
      "THRIVE\n",
      "None\n",
      "None\n",
      "----\n",
      "OmniSearch\n",
      "None\n",
      "None\n",
      "----\n",
      "TCIA\n",
      "None\n",
      "None\n",
      "----\n",
      "HIVE (BioExpress)\n",
      "None\n",
      "None\n",
      "----\n",
      "TCPA\n",
      "None\n",
      "None\n",
      "----\n",
      "TIES\n",
      "None\n",
      "None\n",
      "----\n",
      "[{\"numberVerification\": [{\"longNumber\": 281474976710655}]}, {\"metaData\": [{\"elementCount\": 241, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"edgeAttributes\"}, {\"elementCount\": 3, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"cyVisualProperties\"}, {\"elementCount\": 1, \"lastUpdate\": 1509479242423, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"provenanceHistory\"}, {\"elementCount\": 6, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"networkAttributes\"}, {\"idCounter\": 81, \"elementCount\": 81, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"edges\"}, {\"elementCount\": 36, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"cartesianLayout\"}, {\"elementCount\": 1, \"consistencyGroup\": 1, \"name\": \"@context\"}, {\"idCounter\": 36, \"elementCount\": 36, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"nodes\"}]}, {\"nodes\": [{\"r\": \"C-BIBOP\", \"@id\": 1, \"n\": \"C-BIBOP\"}, {\"r\": \"3D Slicer\", \"@id\": 2, \"n\": \"3D Slicer\"}, {\"r\": \"CaPTK\", \"@id\": 3, \"n\": \"CaPTK\"}, {\"r\": \"DICOM ToolKit (DCMTK)\", \"@id\": 4, \"n\": \"DICOM ToolKit (DCMTK)\"}, {\"r\": \"LesionTracker\", \"@id\": 5, \"n\": \"LesionTracker\"}, {\"r\": \"PETCT-CoSeg\", \"@id\": 6, \"n\": \"PETCT-CoSeg\"}, {\"r\": \"QUIP\", \"@id\": 7, \"n\": \"QUIP\"}, {\"r\": \"SlicerDMRI\", \"@id\": 8, \"n\": \"SlicerDMRI\"}, {\"r\": \"XNAT\", \"@id\": 9, \"n\": \"XNAT\"}, {\"r\": \"GenePattern\", \"@id\": 10, \"n\": \"GenePattern\"}, {\"r\": \"AMARETTO\", \"@id\": 11, \"n\": \"AMARETTO\"}, {\"r\": \"MuPIT\", \"@id\": 12, \"n\": \"MuPIT\"}, {\"r\": \"BMEG\", \"@id\": 13, \"n\": \"BMEG\"}, {\"r\": \"NG-CHM\", \"@id\": 14, \"n\": \"NG-CHM\"}, {\"r\": \"CDSA\", \"@id\": 15, \"n\": \"CDSA\"}, {\"r\": \"dcmqi\", \"@id\": 16, \"n\": \"dcmqi\"}, {\"r\": \"RADIOMICS.io\", \"@id\": 17, \"n\": \"RADIOMICS.io\"}, {\"r\": \"Bioconductor\", \"@id\": 18, \"n\": \"Bioconductor\"}, {\"r\": \"cBioPortal\", \"@id\": 19, \"n\": \"cBioPortal\"}, {\"r\": \"NDEx\", \"@id\": 20, \"n\": \"NDEx\"}, {\"r\": \"CIViC\", \"@id\": 21, \"n\": \"CIViC\"}, {\"r\": \"HIVE (BioMuta)\", \"@id\": 22, \"n\": \"HIVE (BioMuta)\"}, {\"r\": \"PDX Finder\", \"@id\": 23, \"n\": \"PDX Finder\"}, {\"r\": \"Trinity\", \"@id\": 24, \"n\": \"Trinity\"}, {\"r\": \"FHIR ShEx\", \"@id\": 25, \"n\": \"FHIR ShEx\"}, {\"r\": \"CLAMP-Cancer\", \"@id\": 26, \"n\": \"CLAMP-Cancer\"}, {\"r\": \"Galaxy-P multi-omics\", \"@id\": 27, \"n\": \"Galaxy-P multi-omics\"}, {\"r\": \"CRAVAT\", \"@id\": 28, \"n\": \"CRAVAT\"}, {\"r\": \"UCSC Xena\", \"@id\": 29, \"n\": \"UCSC Xena\"}, {\"r\": \"Integrative Genomics Viewer (IGV)\", \"@id\": 30, \"n\": \"Integrative Genomics Viewer (IGV)\"}, {\"r\": \"THRIVE\", \"@id\": 31, \"n\": \"THRIVE\"}, {\"r\": \"OmniSearch\", \"@id\": 32, \"n\": \"OmniSearch\"}, {\"r\": \"TCIA\", \"@id\": 33, \"n\": \"TCIA\"}, {\"r\": \"HIVE (BioExpress)\", \"@id\": 34, \"n\": \"HIVE (BioExpress)\"}, {\"r\": \"TCPA\", \"@id\": 35, \"n\": \"TCPA\"}, {\"r\": \"TIES\", \"@id\": 36, \"n\": \"TIES\"}]}, {\"edges\": [{\"i\": \"Connected with\", \"s\": 1, \"@id\": 1, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 3, \"@id\": 2, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 4, \"@id\": 3, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 5, \"@id\": 4, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 6, \"@id\": 5, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 7, \"@id\": 6, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 8, \"@id\": 7, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 9, \"@id\": 8, \"t\": 2}, {\"i\": \"Connected with\", \"s\": 10, \"@id\": 9, \"t\": 11}, {\"i\": \"Connected with\", \"s\": 12, \"@id\": 10, \"t\": 13}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 11, \"t\": 1}, {\"i\": \"Connected with\", \"s\": 1, \"@id\": 12, \"t\": 3}, {\"i\": \"Connected with\", \"s\": 15, \"@id\": 13, \"t\": 3}, {\"i\": \"Connected with\", \"s\": 16, \"@id\": 14, \"t\": 3}, {\"i\": \"Connected with\", \"s\": 17, \"@id\": 15, \"t\": 3}, {\"i\": \"Connected with\", \"s\": 9, \"@id\": 16, \"t\": 3}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 17, \"t\": 19}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 18, \"t\": 19}, {\"i\": \"Connected with\", \"s\": 20, \"@id\": 19, \"t\": 19}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 20, \"t\": 15}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 21, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 22, \"@id\": 22, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 23, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 24, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 23, \"@id\": 25, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 24, \"@id\": 26, \"t\": 21}, {\"i\": \"Connected with\", \"s\": 25, \"@id\": 27, \"t\": 26}, {\"i\": \"Connected with\", \"s\": 27, \"@id\": 28, \"t\": 28}, {\"i\": \"Connected with\", \"s\": 20, \"@id\": 29, \"t\": 28}, {\"i\": \"Connected with\", \"s\": 29, \"@id\": 30, \"t\": 28}, {\"i\": \"Connected with\", \"s\": 1, \"@id\": 31, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 3, \"@id\": 32, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 4, \"@id\": 33, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 5, \"@id\": 34, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 17, \"@id\": 35, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 9, \"@id\": 36, \"t\": 16}, {\"i\": \"Connected with\", \"s\": 9, \"@id\": 37, \"t\": 4}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 38, \"t\": 30}, {\"i\": \"Connected with\", \"s\": 21, \"@id\": 39, \"t\": 30}, {\"i\": \"Connected with\", \"s\": 27, \"@id\": 40, \"t\": 30}, {\"i\": \"Connected with\", \"s\": 24, \"@id\": 41, \"t\": 30}, {\"i\": \"Connected with\", \"s\": 1, \"@id\": 42, \"t\": 5}, {\"i\": \"Connected with\", \"s\": 3, \"@id\": 43, \"t\": 5}, {\"i\": \"Connected with\", \"s\": 31, \"@id\": 44, \"t\": 5}, {\"i\": \"Connected with\", \"s\": 9, \"@id\": 45, \"t\": 5}, {\"i\": \"Connected with\", \"s\": 21, \"@id\": 46, \"t\": 12}, {\"i\": \"Connected with\", \"s\": 10, \"@id\": 47, \"t\": 12}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 48, \"t\": 12}, {\"i\": \"Connected with\", \"s\": 23, \"@id\": 49, \"t\": 12}, {\"i\": \"Connected with\", \"s\": 10, \"@id\": 50, \"t\": 14}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 51, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 52, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 26, \"@id\": 53, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 30, \"@id\": 54, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 20, \"@id\": 55, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 29, \"@id\": 56, \"t\": 32}, {\"i\": \"Connected with\", \"s\": 19, \"@id\": 57, \"t\": 23}, {\"i\": \"Connected with\", \"s\": 15, \"@id\": 58, \"t\": 23}, {\"i\": \"Connected with\", \"s\": 2, \"@id\": 59, \"t\": 17}, {\"i\": \"Connected with\", \"s\": 15, \"@id\": 60, \"t\": 17}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 61, \"t\": 17}, {\"i\": \"Connected with\", \"s\": 6, \"@id\": 62, \"t\": 17}, {\"i\": \"Connected with\", \"s\": 7, \"@id\": 63, \"t\": 8}, {\"i\": \"Connected with\", \"s\": 2, \"@id\": 64, \"t\": 33}, {\"i\": \"Connected with\", \"s\": 1, \"@id\": 65, \"t\": 33}, {\"i\": \"Connected with\", \"s\": 34, \"@id\": 66, \"t\": 35}, {\"i\": \"Connected with\", \"s\": 14, \"@id\": 67, \"t\": 35}, {\"i\": \"Connected with\", \"s\": 18, \"@id\": 68, \"t\": 20}, {\"i\": \"Connected with\", \"s\": 13, \"@id\": 69, \"t\": 20}, {\"i\": \"Connected with\", \"s\": 26, \"@id\": 70, \"t\": 36}, {\"i\": \"Connected with\", \"s\": 29, \"@id\": 71, \"t\": 36}, {\"i\": \"Connected with\", \"s\": 28, \"@id\": 72, \"t\": 24}, {\"i\": \"Connected with\", \"s\": 10, \"@id\": 73, \"t\": 24}, {\"i\": \"Connected with\", \"s\": 19, \"@id\": 74, \"t\": 29}, {\"i\": \"Connected with\", \"s\": 28, \"@id\": 75, \"t\": 29}, {\"i\": \"Connected with\", \"s\": 35, \"@id\": 76, \"t\": 29}, {\"i\": \"Connected with\", \"s\": 20, \"@id\": 77, \"t\": 29}, {\"i\": \"Connected with\", \"s\": 13, \"@id\": 78, \"t\": 9}, {\"i\": \"Connected with\", \"s\": 3, \"@id\": 79, \"t\": 9}, {\"i\": \"Connected with\", \"s\": 8, \"@id\": 80, \"t\": 9}, {\"i\": \"Connected with\", \"s\": 31, \"@id\": 81, \"t\": 9}]}, {\"networkAttributes\": [{\"v\": \"ITCR Connectivity Map\", \"d\": \"string\", \"n\": \"name\"}, {\"v\": \"<h5>This network shows the interactions between projects funded under the <a href=\\\"https://itcr.cancer.gov/\\\" target=\\\"\\\" style=\\\"font-weight: bold;\\\">Informatics Technology for Cancer Research</a><b>\\u00a0</b>(ITCR) program and the bioinformatic tools they create. Edges in the graph can be selected to show information about tools integration and the status of the collaboration.</h5><pre><b>Edge Color legend</b><br/><span id=\\\"selectionBoundary_1509478417407_8323839425500312\\\" class=\\\"rangySelectionBoundary\\\">&#65279;</span>Orange solid: Existing connections<br/>Blue solid: Ongoing connections<br/>Grey dashed: Proposed connections<span id=\\\"selectionBoundary_1509478417407_8897326232996032\\\" class=\\\"rangySelectionBoundary\\\">&#65279;</span></pre><p></p>\", \"d\": \"string\", \"n\": \"description\"}, {\"v\": \"1.1 (OCT-2017)\", \"d\": \"string\", \"n\": \"version\"}, {\"v\": \"Rudolf T. Pillich, PhD\", \"d\": \"string\", \"n\": \"author\"}, {\"v\": \"NIH, NCI, ITCR\", \"d\": \"string\", \"n\": \"labels\"}, {\"v\": \"Connectivity Map\", \"d\": \"string\", \"n\": \"networkType\"}]}, {\"edgeAttributes\": [{\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 1, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 1, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Quantitative image analysis tools are used in C-BIBOP to implement processing pipelines\", \"po\": 1, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 2, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos\", \"po\": 2, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Make CaPTk functionality available as Slicer Plugin\", \"po\": 2, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 3, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 3, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"DCMTK is a library that implements low-level API for working with DICOM data\", \"po\": 3, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 4, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 4, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate analysis tools implemented in 3D Slicer server-side with the visualization and annotation tools implemented in OHIF LesionTracker\", \"po\": 4, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 5, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 5, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Collection of 3D Slicer extensions for quantitative analysis of PET-CT data\", \"po\": 5, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 6, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Joel Saltz\", \"po\": 6, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Pathology ground truth, 3D Pathology segmentation\", \"po\": 6, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 7, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 7, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"3D Slicer is the interactive platform, where SlicerDMRI is a problem-specific solution\", \"po\": 7, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 8, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 8, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Use 3D Slicer as an interface to interact with the XNAT-hosted data\", \"po\": 8, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 9, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Nathalie Pochet\", \"po\": 9, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Analytic modules in GenePattern to infer & benchmark regulatory networks (virus-induced cancer)\", \"po\": 9, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 10, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Ryan Spangler\", \"po\": 10, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Incorporate MuPIT visualization into BMEG browser's variants page\", \"po\": 10, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 11, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jayashree Kalpathy-Cramer\", \"po\": 11, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualization for radiomics challenges and to develop feature ontology\", \"po\": 11, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 12, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jayashree Kalpathy-Cramer\", \"po\": 12, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate CapTk algorithms into C-BIBOP\", \"po\": 12, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 13, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Gutman/Cooper\", \"po\": 13, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Common python image feature set\", \"po\": 13, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 14, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos\", \"po\": 14, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate dcmqi functionality into CaPTk\", \"po\": 14, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 15, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos\", \"po\": 15, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate radiomic features into CaPTk predictive models\", \"po\": 15, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 16, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos / Spyridon Bakas\", \"po\": 16, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Use CaPTk as an interface to interact with the XNAT-hosted data\", \"po\": 16, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 17, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"JJ Gao\", \"po\": 17, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Bioconductor Package of loading genomics and clinical data from the cBioPortal API\", \"po\": 17, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 18, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Bradley Broom\", \"po\": 18, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Link to cBioPortal from selected samples in NG-CHM\", \"po\": 18, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 19, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Dexter Pratt\", \"po\": 19, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Annotate network nodes with alteration data from cBioPortal\", \"po\": 19, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 20, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Bradley Broom\", \"po\": 20, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualize TCGA slide of sample included in heat map\", \"po\": 20, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 21, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"mtmorgan\", \"po\": 21, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Place statistically supported results into biological context\", \"po\": 21, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 22, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Mazumder/Crichton\", \"po\": 22, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Link from OncoMX mutations to CIViC record, when available\", \"po\": 22, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 23, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Bradley Broom\", \"po\": 23, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Link to CIViC gene record pages from heatmap gene labels\", \"po\": 23, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 24, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Bradley Broom\", \"po\": 24, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"View variant information of genes included in a heat map\", \"po\": 24, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 25, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Provide PDX model/sample variants to CIViC\", \"po\": 25, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 26, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Brian Haas\", \"po\": 26, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"inferring significance of rna-seq based fusions & mutations, also curating fusions in literature\", \"po\": 26, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 27, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Guoqian Jiang\", \"po\": 27, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrating FHIR-based data model validation into the CLAMP pipeline\", \"po\": 27, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 28, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Tim Griffin / Pratik Jagtap\", \"po\": 28, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"For interpretation of potential pathology associated with variant peptide sequences identified via proteogenomics\", \"po\": 28, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 29, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Dexter Pratt\", \"po\": 29, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"NDEx has been incorporated into the CRAVAT application as a source of networks for pathway enrichment analysis of variants.\", \"po\": 29, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 30, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingchun Zhu\", \"po\": 30, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Intepret mutations\", \"po\": 30, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 31, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 31, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"dcmqi provides converters used by C-BIBOP for normalizing data stored in research formats\", \"po\": 31, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 32, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 32, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Conversion of the analysis results into structured and standardized format\", \"po\": 32, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 33, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 33, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"dcmqi provides a set of abstractions to simplify data conversion of specific types of DICOM objects, is based on DCMTK\", \"po\": 33, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 34, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 34, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Provide capability to use LesionTracker to visualize image analysis results converted into standardized DICOM representation using dcmci.\", \"po\": 34, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 35, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 35, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"dcmqi can be used to store the radiomics features in a structured and standardized format\", \"po\": 35, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 36, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 36, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Conversion of the analysis results into structured and standardized format\", \"po\": 36, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 37, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"John Flavin\", \"po\": 37, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integration of DICOM object headers and images into XNAT data model\", \"po\": 37, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 38, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"mtmrogan\", \"po\": 38, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualize (and update) computed results, like epivizr\", \"po\": 38, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 39, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Obi Griffith\", \"po\": 39, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Provide CIViC variants as optional annotation track\", \"po\": 39, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 40, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Tim Griffin / Pratik Jagtap\", \"po\": 40, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"For visualization of known and novel proteoform-specific peptides on the genome\", \"po\": 40, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 41, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Brian Haas / James Robinson\", \"po\": 41, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualize ab initio-assembled transcriptomes, read alignments, sequence variants, and fusions\", \"po\": 41, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 42, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jayashree Kalpathy-Cramer\", \"po\": 42, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate Cornerstone viewer within C-BIBOP\", \"po\": 42, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 43, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos\", \"po\": 43, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate segmentation and radiomic features of CaPTk into LesionTracker\", \"po\": 43, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 44, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 44, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Cornerstone is used for visualizing multiplexed data\", \"po\": 44, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 45, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Dan Marcus\", \"po\": 45, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Emded LesionTracker as visualization/measurement app in XNAT\", \"po\": 45, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 46, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Shows 3D structure location of variant(s)\", \"po\": 46, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 47, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Michael Reich\", \"po\": 47, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Embed 3D visualizations within interactive genomics analysis notebooks\", \"po\": 47, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 48, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Bradley Broom\", \"po\": 48, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"View structure of genes included in a heat map\", \"po\": 48, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 49, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Carol Bult\", \"po\": 49, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"To visualize mutations in context of 3D structures and assess significance\", \"po\": 49, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 50, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Michael Reich\", \"po\": 50, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Enable sophisticated visualization and manipulation of heatmaps as steps within online analytical notebooks\", \"po\": 50, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 51, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingshan Huang\", \"po\": 51, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"A new set of cancer genomic analysis tasks that can be readily integrated with existing Bioconductor tools\", \"po\": 51, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 52, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingshan Huang\", \"po\": 52, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"GEO data from Bioconductor through its AnnotationHub and ExperimentHub\", \"po\": 52, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 53, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Hua Xu\", \"po\": 53, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Use CLAMP to develop literature mining pipelines\", \"po\": 53, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 54, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingshan Huang\", \"po\": 54, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"IGV genomics viewer for OmniSearch search and analysis results\", \"po\": 54, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 55, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingshan Huang\", \"po\": 55, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Identified candidate networks for ncRNAs under investigation\", \"po\": 55, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 56, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingshan Huang\", \"po\": 56, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"TCGA and GTEx data provided by Xena Data Store\", \"po\": 56, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 57, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"JJ Gao\", \"po\": 57, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Cross-linking the PDX samples in the two resources\", \"po\": 57, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 58, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Gutman/Cooper\", \"po\": 58, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Mouse phenotype histology portal\", \"po\": 58, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 59, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 59, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Interactive interface to the command-line/batch processing tools\", \"po\": 59, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 60, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Gutman/Cooper\", \"po\": 60, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Common python image feature set\", \"po\": 60, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 61, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Juli Klemm\", \"po\": 61, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualize radiomics data with clustered heat maps\", \"po\": 61, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 62, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Xiaodong Wu\", \"po\": 62, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate to the radiomics tools to facilate the use of both PET and CT\", \"po\": 62, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 63, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Lauren O'Donnell\", \"po\": 63, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Evaluate dMRI microstructure measures vs stereotactic samples from neurosurgery\", \"po\": 63, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 64, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Andrey Fedorov\", \"po\": 64, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Use 3D Slicer as an interface to access TCIA-hosted data\", \"po\": 64, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 65, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jayashree Kalpathy-Cramer\", \"po\": 65, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Algorithm comparison on data hosted in TCIA\", \"po\": 65, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 66, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Mazumder/Crichton\", \"po\": 66, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Map to Cancer-DO Slim and identify genes that have concordance of RNA-seq and protein chip data across cancers\", \"po\": 66, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 67, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Juli Klemm\", \"po\": 67, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Visualization of RPPA data\", \"po\": 67, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Exists\", \"po\": 68, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Rudi Pillich\", \"po\": 68, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"R client available as a Bioconductor package for programmatic access to the NDEx REST API\", \"po\": 68, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 69, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Josh Stuart\", \"po\": 69, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Ingest network models into evidence graph\", \"po\": 69, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 70, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Hua Xu\", \"po\": 70, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Use CLAMP NLP modules for TIES\", \"po\": 70, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 71, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Juli Klemm\", \"po\": 71, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate TIES's NLP-based clinical data search with the Xena browser allowing Xena users to use clinical report text to search for samples and genomic results.\", \"po\": 71, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Ongoing\", \"po\": 72, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Brian Haas\", \"po\": 72, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Variant annotation - 3D structure with variants\", \"po\": 72, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 73, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Brian Haas\", \"po\": 73, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Enable running Trinity CTAT modules from GenePattern\", \"po\": 73, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 74, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"JJ Gao\", \"po\": 74, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Co-develop Javascript lib for survival analysis\", \"po\": 74, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 75, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingchun Zhu\", \"po\": 75, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Dynamic annotate mutations\", \"po\": 75, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 76, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Mary Goldman\", \"po\": 76, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Add RPPA data for CCLE data (RNAseq and CNV) already in Xena\", \"po\": 76, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 77, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Jingchun Zhu\", \"po\": 77, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Display genomics data in the context of biological pathways\", \"po\": 77, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 78, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Juli Klemm\", \"po\": 78, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Connect radiomics data with other biomedical evidence\", \"po\": 78, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 79, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Christos Davatzikos\", \"po\": 79, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Integrate CaPTk pipelines into XNAT workflows\", \"po\": 79, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 80, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Lauren O'Donnell\", \"po\": 80, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Enable DICOM tractography in XNAT\", \"po\": 80, \"n\": \"Why?\"}, {\"d\": \"string\", \"v\": \"Proposed\", \"po\": 81, \"n\": \"Status\"}, {\"d\": \"string\", \"v\": \"Brion Sarachan\", \"po\": 81, \"n\": \"Suggested By\"}, {\"d\": \"string\", \"v\": \"Workflow to curate pathology image submissions\", \"po\": 81, \"n\": \"Why?\"}]}, {\"cyVisualProperties\": [{\"properties_of\": \"network\", \"properties\": {\"NETWORK_DEPTH\": \"0.0\", \"NETWORK_HEIGHT\": \"400.0\", \"NETWORK_EDGE_SELECTION\": \"true\", \"NETWORK_BACKGROUND_PAINT\": \"#FFFFFF\", \"NETWORK_CENTER_X_LOCATION\": \"0.0\", \"NETWORK_WIDTH\": \"550.0\", \"NETWORK_SIZE\": \"550.0\", \"NETWORK_CENTER_Z_LOCATION\": \"0.0\", \"NETWORK_SCALE_FACTOR\": \"1.0\", \"NETWORK_CENTER_Y_LOCATION\": \"0.0\", \"NETWORK_NODE_SELECTION\": \"true\"}}, {\"properties_of\": \"nodes:default\", \"dependencies\": {\"nodeCustomGraphicsSizeSync\": \"true\", \"nodeSizeLocked\": \"false\"}, \"properties\": {\"NODE_FILL_COLOR\": \"#EFF0F2\", \"NODE_NESTED_NETWORK_IMAGE_VISIBLE\": \"true\", \"NODE_LABEL_TRANSPARENCY\": \"255\", \"NODE_LABEL_FONT_SIZE\": \"16\", \"NODE_CUSTOMPAINT_1\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_1, name=Node Custom Paint 1)\", \"NODE_BORDER_WIDTH\": \"3.0\", \"NODE_CUSTOMPAINT_3\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_3, name=Node Custom Paint 3)\", \"NODE_CUSTOMPAINT_2\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_2, name=Node Custom Paint 2)\", \"NODE_CUSTOMPAINT_5\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_5, name=Node Custom Paint 5)\", \"NODE_CUSTOMPAINT_4\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_4, name=Node Custom Paint 4)\", \"NODE_CUSTOMPAINT_7\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_7, name=Node Custom Paint 7)\", \"NODE_CUSTOMPAINT_6\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_6, name=Node Custom Paint 6)\", \"NODE_CUSTOMPAINT_9\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_9, name=Node Custom Paint 9)\", \"NODE_CUSTOMPAINT_8\": \"DefaultVisualizableVisualProperty(id=NODE_CUSTOMPAINT_8, name=Node Custom Paint 8)\", \"NODE_LABEL_COLOR\": \"#000000\", \"NODE_SELECTED\": \"false\", \"NODE_WIDTH\": \"130.0\", \"NODE_SHAPE\": \"ELLIPSE\", \"NODE_SIZE\": \"35.0\", \"NODE_BORDER_TRANSPARENCY\": \"255\", \"NODE_CUSTOMGRAPHICS_SIZE_8\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_9\": \"50.0\", \"NODE_SELECTED_PAINT\": \"#FF3333\", \"NODE_LABEL_POSITION\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_SIZE_1\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_2\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_3\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_4\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_5\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_6\": \"50.0\", \"NODE_CUSTOMGRAPHICS_SIZE_7\": \"50.0\", \"NODE_BORDER_PAINT\": \"#666666\", \"NODE_LABEL_FONT_FACE\": \"Dialog,plain,12\", \"NODE_CUSTOMGRAPHICS_POSITION_2\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_3\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_1\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_6\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_7\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_4\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_5\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_8\": \"C,C,c,0.00,0.00\", \"NODE_CUSTOMGRAPHICS_POSITION_9\": \"C,C,c,0.00,0.00\", \"NODE_X_LOCATION\": \"0.0\", \"NODE_Y_LOCATION\": \"0.0\", \"NODE_DEPTH\": \"0.0\", \"NODE_LABEL_WIDTH\": \"120.0\", \"COMPOUND_NODE_SHAPE\": \"ROUND_RECTANGLE\", \"NODE_CUSTOMGRAPHICS_9\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_8\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_5\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_4\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_BORDER_STROKE\": \"SOLID\", \"NODE_CUSTOMGRAPHICS_6\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_1\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_3\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_CUSTOMGRAPHICS_2\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_HEIGHT\": \"70.0\", \"NODE_Z_LOCATION\": \"0.0\", \"NODE_PAINT\": \"#1E90FF\", \"NODE_TRANSPARENCY\": \"200\", \"COMPOUND_NODE_PADDING\": \"10.0\", \"NODE_CUSTOMGRAPHICS_7\": \"org.cytoscape.ding.customgraphics.NullCustomGraphics,0,[ Remove Graphics ],\", \"NODE_VISIBLE\": \"true\"}, \"mappings\": {\"NODE_LABEL\": {\"definition\": \"COL=name,T=string\", \"type\": \"PASSTHROUGH\"}}}, {\"properties_of\": \"edges:default\", \"dependencies\": {\"arrowColorMatchesEdge\": \"true\"}, \"properties\": {\"EDGE_PAINT\": \"#323232\", \"EDGE_TARGET_ARROW_UNSELECTED_PAINT\": \"#000000\", \"EDGE_CURVED\": \"true\", \"EDGE_TARGET_ARROW_SELECTED_PAINT\": \"#FFFF00\", \"EDGE_STROKE_SELECTED_PAINT\": \"#FF3333\", \"EDGE_LABEL_FONT_FACE\": \"Dialog,plain,10\", \"EDGE_LINE_TYPE\": \"SOLID\", \"EDGE_TARGET_ARROW_SHAPE\": \"NONE\", \"EDGE_LABEL_COLOR\": \"#000000\", \"EDGE_SELECTED\": \"false\", \"EDGE_SELECTED_PAINT\": \"#FF0000\", \"EDGE_VISIBLE\": \"true\", \"EDGE_LABEL_FONT_SIZE\": \"10\", \"EDGE_STROKE_UNSELECTED_PAINT\": \"#848484\", \"EDGE_WIDTH\": \"2.0\", \"EDGE_SOURCE_ARROW_SHAPE\": \"NONE\", \"EDGE_LABEL_TRANSPARENCY\": \"255\", \"EDGE_LABEL_WIDTH\": \"200.0\", \"EDGE_UNSELECTED_PAINT\": \"#404040\", \"EDGE_SOURCE_ARROW_UNSELECTED_PAINT\": \"#000000\", \"EDGE_TRANSPARENCY\": \"255\", \"EDGE_SOURCE_ARROW_SELECTED_PAINT\": \"#FFFF00\"}, \"mappings\": {\"EDGE_UNSELECTED_PAINT\": {\"definition\": \"COL=Status,T=string,K=0=Exists,V=0=#FF6600,K=1=Proposed,V=1=#666666,K=2=Ongoing,V=2=#006699\", \"type\": \"DISCRETE\"}, \"EDGE_WIDTH\": {\"definition\": \"COL=Status,T=string,K=0=Exists,V=0=10.0,K=1=Proposed,V=1=4.0,K=2=Ongoing,V=2=6.0\", \"type\": \"DISCRETE\"}, \"EDGE_LINE_TYPE\": {\"definition\": \"COL=Status,T=string,K=0=Exists,V=0=SOLID,K=1=Proposed,V=1=EQUAL_DASH,K=2=Ongoing,V=2=SOLID\", \"type\": \"DISCRETE\"}, \"EDGE_STROKE_UNSELECTED_PAINT\": {\"definition\": \"COL=Status,T=string,K=0=Exists,V=0=#FF9900,K=1=Proposed,V=1=#666666,K=2=Ongoing,V=2=#009900\", \"type\": \"DISCRETE\"}}}]}, {\"@context\": [{}]}, {\"cartesianLayout\": [{\"node\": 36, \"x\": 114.8187255859375, \"y\": -533.8399658203125}, {\"node\": 35, \"x\": -547.3665771484375, \"y\": -14.970279693603516}, {\"node\": 34, \"x\": -484.9322783120931, \"y\": -404.7746276855469}, {\"node\": 33, \"x\": -596.9725341796875, \"y\": 275.13925489103195}, {\"node\": 32, \"x\": 697.5767211914062, \"y\": -232.42398071289062}, {\"node\": 31, \"x\": 549.085205078125, \"y\": 422.76776123046875}, {\"node\": 30, \"x\": -11.463641478312034, \"y\": -449.8281218179524}, {\"node\": 29, \"x\": 260.55096435546875, \"y\": -431.455078125}, {\"node\": 28, \"x\": -200.6170654296875, \"y\": -332.37306784643533}, {\"node\": 27, \"x\": -163.9130859375, \"y\": -538.4292265542805}, {\"node\": 26, \"x\": 549.4385038980306, \"y\": -380.2201506265462}, {\"node\": 25, \"x\": 421.1508757242024, \"y\": -537.2540283203125}, {\"node\": 24, \"x\": -340.63576952303055, \"y\": -503.9375}, {\"node\": 23, \"x\": 218.97623635278342, \"y\": -122.67657788907884}, {\"node\": 22, \"x\": -768.9812285074055, \"y\": -94.50967725431322}, {\"node\": 21, \"x\": -370.68853759765625, \"y\": -204.29139709472656}, {\"node\": 20, \"x\": 53.22990647913825, \"y\": -201.78958129882812}, {\"node\": 19, \"x\": 699.1300048828125, \"y\": -91.12675476074219}, {\"node\": 18, \"x\": 284.4632326824546, \"y\": -281.46307373046875}, {\"node\": 17, \"x\": 257.60226122224924, \"y\": 20.823219299316406}, {\"node\": 16, \"x\": -0.9302291870117188, \"y\": 546.6609834066569}, {\"node\": 15, \"x\": 656.173095703125, \"y\": 19.43295669555664}, {\"node\": 14, \"x\": -21.56396484375, \"y\": -87.23457336425781}, {\"node\": 13, \"x\": -290.4934113851726, \"y\": -108.89071210538744}, {\"node\": 12, \"x\": -530.860595703125, \"y\": 163.7381134033203}, {\"node\": 11, \"x\": -606.2202485433758, \"y\": -299.3152160644531}, {\"node\": 10, \"x\": -718.8748168945312, \"y\": 90.58221435546875}, {\"node\": 9, \"x\": 33.33720397949219, \"y\": 38.43511962890625}, {\"node\": 8, \"x\": -157.38243103027344, \"y\": 442.141357421875}, {\"node\": 7, \"x\": -398.3072814941406, \"y\": 529.068508780843}, {\"node\": 6, \"x\": 570.0255737304688, \"y\": 310.18780517578125}, {\"node\": 5, \"x\": 338.6950346597493, \"y\": 498.34478759765625}, {\"node\": 4, \"x\": -567.1018676757812, \"y\": 407.24188232421875}, {\"node\": 3, \"x\": 563.0072631835938, \"y\": 119.95858764648438}, {\"node\": 2, \"x\": -272.854248046875, \"y\": 311.42657470703125}, {\"node\": 1, \"x\": -251.573974609375, \"y\": 139.1765899658203}]}, {\"metaData\": [{\"elementCount\": 241, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"edgeAttributes\"}, {\"elementCount\": 3, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"cyVisualProperties\"}, {\"elementCount\": 1, \"lastUpdate\": 1509479242423, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"provenanceHistory\"}, {\"elementCount\": 6, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"networkAttributes\"}, {\"idCounter\": 81, \"elementCount\": 81, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"edges\"}, {\"elementCount\": 36, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"cartesianLayout\"}, {\"elementCount\": 1, \"consistencyGroup\": 1, \"name\": \"@context\"}, {\"idCounter\": 36, \"elementCount\": 36, \"version\": \"1.0\", \"consistencyGroup\": 1, \"name\": \"nodes\"}]}]\n"
     ]
    }
   ],
   "source": [
    "counter = 0\n",
    "for id, node in itcr_network.get_nodes():\n",
    "    if counter > 3:\n",
    "        break;\n",
    "    print(node.get_name())\n",
    "    print(itcr_network.get_node_attribute(node, \"label\"))\n",
    "    print(itcr_network.get_node_attribute(node, \"Tool Description\"))\n",
    "    print(\"----\")\n",
    "print(itcr_network)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Upload the network to your NDEx account as a new network.\n",
    "\n",
    "If you started by loading an one of your networks (such as a clone of the example network), then you can *update* it instead, overwriting its content. \n",
    "\n",
    "In the commented line of below, the optional parameter *update_uuid* specifies the update behavior."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "upload_message = my_network.upload(my_server, my_account, my_password)\n",
    "# upload_message = my_network.upload(my_server, my_account, my_password, update_uuid=my_network_uuid)\n",
    "print(upload_message)"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
